coldfront/core/user/admin.py

```diff
@@ -17,3 +19,38 @@ def first_name(self, obj):
 
     def last_name(self, obj):
         return obj.user.last_name
+
+
+@admin.register(EmailAddress)
+class EmailAddressAdmin(admin.ModelAdmin):
+    list_display = ('user', 'email', 'is_primary', 'is_verified', )
+    ordering = ('user', '-is_primary', '-is_verified', 'email', )
+    list_filter = ('is_primary', 'is_verified', )
+    search_fields = ('user', 'email', )
+    fields = ('email', 'user', 'is_primary', 'is_verified', )
+    actions = ('make_primary', )
+    readonly_fields = ('is_primary', 'is_verified', )
+
+    # custom logic for delete button in emailaddress admin form
+    def delete_model(self, request, obj):
+        if obj.is_primary:
+            raise ValidationError('Cannot delete primary email. Unset primary status in list display before deleting.')
+        else:
+            super().delete_model(request, obj)
+
+    # custom action to give an email primary status without causing conflicting states
+    @admin.action(description='Make selected primary email')
+    def make_primary(self, request, queryset):
```

Comment:
The input here is a `queryset`. I think it'd be appropriate to raise a `ValidationError` or similar exception if it contains more than one selection (`queryset.count() > 1`).

This way, we're not performing unnecessary work for the first `n - 1`, when only the last one is going to get set in the end.

Reply:
Do we want admins to have the ability to set the primary email for multiple users at once?

Reply:
@matthew-li

Reply:
No, I don't see a need for that.

Reply:
@jofeinstein Okay, I was thinking about the case when an admin selects multiple `EmailAddress` instances that belong to the same user. In that scenario, only the last one would end up getting set, so we'd be wasting computation.

You're talking about addresses that belong to different users.

Let's stick with limiting the admin to selecting only one, for
(a) the reason above, and
(b): _in the future_, any change to a user's primary email address will need to be reflected in the cluster. An admin could accidentally select all, and wind up creating a lot of changes.